### PR TITLE
DB: Deprecate - set_is_elementor_page

### PR DESCRIPTION
**Pull request by Nevoss**

### CODE REVIEW

File: includes/db.php [hunk 1]
```diff
@@ -160,19 +160,30 @@ public function get_new_editor_from_wp_editor( $post_id ) {
...
 	 *                           Default is true.
 	 */
 	public function set_is_elementor_page( $post_id, $is_elementor = true ) {
-		if ( $is_elementor ) {
-			// Use the string `builder` and not a boolean for rollback compatibility
-			update_post_meta( $post_id, '_elementor_edit_mode', 'builder' );
-		} else {
-			delete_post_meta( $post_id, '_elementor_edit_mode' );
+		Plugin::$instance->modules_manager
+			->get_modules('dev-tools')
```
Comment (rodolphebertozzo) [inline includes/db.php:172]: ->get_modules( 'dev-tools' )

File: core/admin/admin.php [hunk 2]
```diff
@@ -199,7 +199,7 @@ public function save_post( $post_id ) {
 			return;
 		}
 
-		Plugin::$instance->db->set_is_elementor_page( $post_id, ! empty( $_POST['_elementor_post_mode'] ) );
+		Plugin::$instance->documents->get( $post_id )->set_is_build_with_elementor( ! empty( $_POST['_elementor_post_mode'] ) );
```
Comment (matipojo) [inline core/admin/admin.php:202]:
```suggestion
		Plugin::$instance->documents->get( $post_id )->set_is_built_with_elementor( ! empty( $_POST['_elementor_post_mode'] ) );
```

File: core/base/document.php [hunk 3]
```diff
@@ -34,6 +34,8 @@ abstract class Document extends Controls_Stack {
 	const TYPE_META_KEY = '_elementor_template_type';
 	const PAGE_META_KEY = '_elementor_page_settings';
 
+	const BUILD_WITH_ELEMENTOR_META_KEY = '_elementor_edit_mode';
```
Comment (matipojo) [inline core/base/document.php:37]:
```suggestion
	const BUILT_WITH_ELEMENTOR_META_KEY = '_elementor_edit_mode';
```

**Review by matipojo — CHANGES_REQUESTED** [review]

File: tests/phpunit/elementor/modules/gutenberg/test-module.php [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+		// Run reset api.
+		$request = new \WP_REST_Request( 'POST', "/wp/v2/posts/{$post->ID}");
+
+		$request->set_body_params([
+			'gutenberg_elementor_mode' => false,
+		]);
+
+		rest_do_request( $request );
+
+		$this->assertEmpty(get_post_meta($post->ID, Document::BUILT_WITH_ELEMENTOR_META_KEY, true));
```
Comment (matipojo) [inline tests/phpunit/elementor/modules/gutenberg/test-module.php:33]: PHPCS

**Review by matipojo — CHANGES_REQUESTED** [review]

File: tests/phpunit/elementor/modules/history/test-revisions-manager.php [hunk 5]
```diff
@@ -147,12 +148,13 @@ public function test_should_restore_revision() {
...
 		$autosave_post_id = $res['child_id'];
 
-		update_metadata( 'post', $autosave_post_id, '_elementor_edit_mode', true );
+		update_metadata( 'post', $autosave_post_id, Document::BUILT_WITH_ELEMENTOR_META_KEY, 'builder' );
 		update_metadata( 'post', $autosave_post_id, '_elementor_meta_data', 'content' );
 
 		Revisions_Manager::restore_revision( $post_id, $autosave_post_id );
 
 		$this->assertTrue( $this->check_revisions( $post_id, $autosave_post_id ) );
+		$this->assertEquals('builder', get_post_meta($post_id, Document::BUILT_WITH_ELEMENTOR_META_KEY, true));
```
Comment (matipojo) [inline tests/phpunit/elementor/modules/history/test-revisions-manager.php:157]: PHPCS

**Review by matipojo — CHANGES_REQUESTED** [review]

**Review by matipojo — APPROVED** [review]

File: core/base/document.php [hunk 6]
```diff
@@ -623,7 +625,25 @@ public function update_settings( array $new_settings ) {
...
+	 * @return $this
+	 */
+	public function set_is_built_with_elementor( $is_built_with_elementor ) {
+		if ( $is_built_with_elementor ) {
+			// Use the string `builder` and not a boolean for rollback compatibility
+			update_post_meta( $this->post->ID, self::BUILT_WITH_ELEMENTOR_META_KEY, 'builder' );
+		} else {
+			delete_post_meta( $this->post->ID, self::BUILT_WITH_ELEMENTOR_META_KEY );
+		}
+
```
Comment (kobizz) [inline core/base/document.php:645]: Use the designated `Document::update_meta` and `Document::delete_meta` methods

File: core/base/document.php [hunk 7]
```diff
@@ -623,7 +625,25 @@ public function update_settings( array $new_settings ) {
 	 * @return bool Whether the post was built with Elementor.
 	 */
 	public function is_built_with_elementor() {
-		return ! ! get_post_meta( $this->post->ID, '_elementor_edit_mode', true );
+		return ! ! get_post_meta( $this->post->ID, self::BUILT_WITH_ELEMENTOR_META_KEY, true );
```
Comment (kobizz) [inline core/base/document.php:628]: Use the designated `Document::get_meta` method

**Review by kobizz — CHANGES_REQUESTED** [review]

File: tests/phpunit/elementor/core/admin/test-admin.php [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+		$post = $this->factory()->create_and_get_custom_post( [] );
+
+		// Act
+		$admin->save_post( $post->ID );
+
+		// Assert
+		$this->assertEmpty( get_post_meta( $post->ID, Document::BUILT_WITH_ELEMENTOR_META_KEY, true ) );
+
+		// Arrange
+		$_POST['_elementor_edit_mode_nonce'] = wp_create_nonce( 'admin.php' );
```
Comment (matipojo) [inline tests/phpunit/elementor/core/admin/test-admin.php:25]: Separate to another test and another post.

**Review by matipojo — CHANGES_REQUESTED** [review]

**Review by matipojo — DISMISSED** [review]

**Review by kobizz — DISMISSED** [review]

**Review by matipojo — DISMISSED** [review]

**Review by rodolphebertozzo — DISMISSED** [review]

**Review by KingYes — APPROVED** [review]

**Pull request merged**